lib/attributor/attribute.rb

```diff
@@ -175,6 +186,11 @@ def as_json_schema(shallow: true, example: nil)
       # TODO: not sure if that's correct (we used to get it from the described hash...
       description[:example] = self.dump(example) if example
 
+      # add custom options as x-optionname
+      self.class.custom_options.each do |name, _|
+        description["x-#{name}".to_sym] = self.options[name] if self.options.key?(name)
```

Comment:
same here...i.e., only generate the x- entry if the custom option does not exist in the type/attr  itself?